pkg/server/plugin/upstreamca/spire/spire.go

```diff
@@ -0,0 +1,184 @@
+package spireplugin
+
+import (
+	"context"
+	"crypto/x509"
+	"errors"
+	"fmt"
+	"sync"
+	//"time"
+
+	"github.com/hashicorp/hcl"
+
+	"github.com/spiffe/spire/pkg/common/x509svid"
+	"github.com/spiffe/spire/pkg/common/x509util"
+	"github.com/spiffe/spire/proto/server/upstreamca"
+	"github.com/spiffe/spire/api/workload"
+	"github.com/spiffe/spire/proto/common/plugin"
+
+	"crypto/ecdsa"
+	"github.com/spiffe/spire/pkg/common/bundleutil"
+	"google.golang.org/grpc"
+	"github.com/spiffe/go-spiffe/uri"
+	"github.com/spiffe/spire/pkg/common/idutil"
+	"net/url"
+)
+
+type Configuration struct {
+	TTL             string `hcl:"ttl" json:"ttl"` // time to live for generated certs
+	ServerAddr      string `hcl:"server_address" json:"server_address"`
+	ServerPort      string `hcl:"server_port" json:"server_port"`
+	ServerAgentAddr string `hcl:"server_agent_address" json:"server_agent_address"`
+}
+
+type spirePlugin struct {
+	serialNumber      x509util.SerialNumber
+	workloadApiClient workload.X509Client
+
+	mtx               sync.RWMutex
+	cert              *x509.Certificate
+	upstreamCA        *x509svid.UpstreamCA
+
+	workloadKey       []byte
+	workloadCert      []byte
+	workloadBundle    []byte
+
+	spiffeID          string
+	trustDomain       url.URL
+	privateKey        *ecdsa.PrivateKey
+	intermediateCA    []*x509.Certificate
+	bundle            *bundleutil.Bundle
+
+	conn              *grpc.ClientConn
+	config 		  *Configuration
+}
+
+func (m *spirePlugin) Configure(ctx context.Context, req *plugin.ConfigureRequest) (*plugin.ConfigureResponse, error) {
+	// Parse HCL config payload into config struct
+	config := Configuration{}
+	if err := hcl.Decode(&config, req.Configuration); err != nil {
+		return nil, err
+	}
+
+
+	// fetch certificates and private key to be used for communication to node api
+	if req.GlobalConfig == nil {
+		return nil, errors.New("global configuration is required")
+	}
+
+	if req.GlobalConfig.TrustDomain == "" {
+		return nil, errors.New("trust_domain is required")
+	}
+
+	fmt.Println("[plugin][upstreamca] configured values [%v]",config)
+	fmt.Println("[plugin][upstreamca] global config [%v]",req.GlobalConfig)
+	fmt.Println("[plugin][upstreamca] global config trust domain [%v]",req.GlobalConfig.TrustDomain)
```

Comment:
I assume these will be removed at some point?

We don't handle plugin logging very well at the moment - if it's absolutely required, then we should probably pass a logger in instead of using `fmt`

Reply:
Do you think, we could consider the logger for plugins as low prio issue and some one can pick it. Yes, I hope these fmt are due to WIP.

Reply:
For now , there is no logger implemented in this PR .  A new issue will be created once the PR is merged .

Reply:
Sorry @skdwriting and @trilokgm I missed this one... Yes we can push logging to a later PR, so this is looking good now